hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -372,6 +379,10 @@ public static Map<MetadataPartitionType, HoodieData<HoodieRecord>> convertMetada
               dataMetaClient, isColumnStatsIndexEnabled, columnStatsIndexParallelism, targetColumnsForColumnStatsIndex);
       partitionToRecordsMap.put(MetadataPartitionType.COLUMN_STATS, metadataColumnStatsRDD);
     }
+    if (enabledPartitionTypes.contains(MetadataPartitionType.PARTITION_STATS)) {
+      final HoodieData<HoodieRecord> partitionStatsRDD = convertMetadataToPartitionStatsRecords(commitMetadata, context, dataMetaClient, metadataConfig);
+      partitionToRecordsMap.put(MetadataPartitionType.PARTITION_STATS, partitionStatsRDD);
+    }
```

Comment:
A side topic, does this compose a sub-graph in DAG without triggering the execution?  Ideally, all types of metadata should be computed in parallel leveraging the parallelism in Spark, instead of being computed type by type sequentially.

Reply:
No. That's a very good point! Typically, we are enabling indexes one at a time. But, I agree with you. I will work on it in HUDI-7690. Btw, this limitation is only for initialization. In case of update, we don't compute sequuentially.